### PR TITLE
Check for Site ID when granting access

### LINKED ISSUE #98: pressable-grant-access doesn't work with Site ID or production URL
While trying to set myself up as a collaborator, I had trouble picking the right parameter for the `--site` parameter. Working with the [Tumblr Shop](https://my.pressable.com/sites/1236569), the following commands returned `Site not found.`:

`team51 pressable-grant-access --email=mike.straw@automattic.com --site=1236569`
`team51 pressable-grant-access --email=mike.straw@automattic.com --site=shop.tumblr.com`

The command that did work was:
`team51 pressable-grant-access --email=mike.straw@automattic.com --site=tumblr-swag-store-production.mystagingwebsite.com`

Working with @pippercameron, we found that the site ID isn't checked at all and the [site list](https://github.com/Automattic/team51-cli/blob/54f13adfacad04fc7408ac52439014777afd238d/src/commands/pressable-grant-access.php#L297) doesn't return mapped domain(s).

It's possible we need to just update the [`site` parameter doc](https://github.com/Automattic/team51-cli/blob/54f13adfacad04fc7408ac52439014777afd238d/src/commands/pressable-grant-access.php#L35) to reflect what it actually checks. Or, we could add the capability to check by site ID and/or mapped domain.

### PR DESCRIPTION
**Pull request by GeoJunkie**
Fixes #98

### CODE REVIEW

File: src/commands/pressable-grant-access.php [hunk 1]
```diff
@@ -286,6 +290,19 @@ private function search_for_site_url( string $site_url ): ?\stdClass {
 		return null;
 	}
 
+	/**
+	 * Search for a site by ID
+	 */
+	private function get_site_by_id( int $site_id ): ?\stdClass {
+		$site = $this->api_helper->call_pressable_api( "sites/$site_id", 'GET', array() );
+		// If we have no results, return an empty null.
+		if ( is_null( $site ) || empty( $site->data ) ) {
```
Comment (pippercameron) [inline src/commands/pressable-grant-access.php:299]: Using `empty()` will cover null states also, so no need for the `is_null()` check!

**Review by pippercameron — CHANGES_REQUESTED** [review]

File: src/commands/pressable-grant-access.php (hunk 1 above)
Comment (pippercameron) [reply src/commands/pressable-grant-access.php:299]: Ah, yes. I understand your thinking there. A few things to consider:

* The `call_pressable_api()` function could return a JSON encoded error message (or some other unexpected non-null value JSON encoded data). Null is returned if `json_encode()` fails to encode what's passed to it, which is a limited case to cover and if it evaluates to true, it doesn't tell us `$site` is anything we can actually use (or even if it's an object).
* Internally, `empty()` does an `isset()` first (on the object property in this case), so all you need is `empty( $site->data )` to safely evaluate whether `$site->data` contains data that doesn't evaluate to empty.

Comment (GeoJunkie) [reply src/commands/pressable-grant-access.php:299]: > Internally, empty() does an isset() first (on the object property in this case)

Well, that's a big TIL for me! I'll fix that here and in `pressable_sites()`

**Review by pippercameron — APPROVED** [review]
Nice!

**Pull request merged**